test/arcade/core/util/GraphTest.java

```diff
@@ -0,0 +1,498 @@
+        Graph graph = new Graph();
+        Node node1 = new Node(0, 0, 0);
+        Node node2 = new Node(0, 1, 0);
+        Edge edge = new Edge(node1, node2);
+
+        graph.addEdge(edge);
+        graph.reverseEdge(edge);
+
+        assertTrue(graph.contains(new Edge(node2, node1)));
+    }
+
+    @Test
+    public void reverseEdge_called_preservesOtherEdges() {
+        Graph graph = new Graph();
+        Node node1 = new Node(0, 0, 0);
+        Node node2 = new Node(0, 1, 0);
+        Node node3 = new Node(0, 2, 0);
+        Edge edge1 = new Edge(node1, node2);
+        Edge edge2 = new Edge(node2, node3);
+
+        graph.addEdge(edge1);
+        graph.addEdge(edge2);
+        graph.reverseEdge(edge2);
+
+        assertTrue(graph.contains(edge1));
+    }
+
+    @Test
+    public void reverseEdge_called_preservesNodes() {
+        Graph graph = new Graph();
+        Node node1 = new Node(0, 0, 0);
+        Node node2 = new Node(0, 1, 0);
+        Node node3 = new Node(0, 2, 0);
+        Edge edge1 = new Edge(node1, node2);
+        Edge edge2 = new Edge(node2, node3);
+
+        graph.addEdge(edge1);
+        graph.addEdge(edge2);
+        graph.reverseEdge(edge2);
+
+        assertTrue(graph.contains(node1));
+        assertTrue(graph.contains(node2));
+        assertTrue(graph.contains(node3));
+    }
+
+    @Test
+    public void reverseEdge_called_preservesNodeCoordinates() {
+        Graph graph = new Graph();
+        Node node1 = new Node(0, 0, 0);
+        Node node2 = new Node(1, 1, 0);
+        Node node3 = new Node(2, 2, 0);
+        Edge edge1 = new Edge(node1, node2);
+        Edge edge2 = new Edge(node2, node3);
+
+        graph.addEdge(edge1);
+        graph.addEdge(edge2);
+        graph.reverseEdge(edge2);
+
+        assertTrue(graph.contains(new Node(0, 0, 0)));
+        assertTrue(graph.contains(new Node(1, 1, 0)));
+        assertTrue(graph.contains(new Node(2, 2, 0)));
+    }
+
+    @Test
+    public void mergeNodes_called_mergesNodes() {
+        Graph graph = new Graph();
+        Node node1 = new Node(0, 0, 0);
+        Node node2A = new Node(1, 1, 0);
+        Node node2B = new Node(1, 1, 0);
+        Node node3 = new Node(2, 2, 0);
+
+        Edge edge1 = new Edge(node1, node2A);
+        Edge edge2 = new Edge(node2B, node3);
+
+        graph.addEdge(edge1);
+        graph.addEdge(edge2);
+        graph.mergeNodes();
+
+        assertTrue(edge1.getTo() == edge2.getFrom());
+    }
+
+    @Test
+    public void getEdgesOut_called_getsEdgesOut() {
+        Graph graph = new Graph();
+        Node node1 = new Node(0, 0, 0);
+        Node node2 = new Node(1, 1, 0);
+        Node node3 = new Node(2, 2, 0);
+        Edge edge1 = new Edge(node1, node2);
+        Edge edge2 = new Edge(node2, node3);
+
+        graph.addEdge(edge1);
+        graph.addEdge(edge2);
+
+        assertTrue(graph.getEdgesOut(new Node(1, 1, 0)).contains(edge2));
+    }
+
+    @Test
+    public void getEdgesOut_called_returnsNull() {
```

Comment:
You could consider changing "called" in the middle of your test names to describe the input for tests of functions where you test multiple input circumstances. So here maybe "calledOnDisconnectedNode" while the test above is "calledOnConnectedNode" or something.